tests/test_device_update.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+import serial
+from pydantic import ValidationError
+from unittest.mock import MagicMock, patch, call
+from miniscope_io.models.devupdate import UpdateCommandDefinitions, UpdateTarget
+from miniscope_io.device_update import DevUpdate, find_ftdi_device
+
+
+@patch("miniscope_io.device_update.serial.tools.list_ports.comports")
+@patch("miniscope_io.device_update.serial.Serial")
+def test_devupdate_with_device_connected(mock_serial, mock_comports):
+    """
+    Test DevUpdate function with a device connected.
+    """
+    mock_serial_instance = mock_serial.return_value
+    mock_comports.return_value = [
+        MagicMock(vid=0x0403, pid=0x6001, device="COM3"),
+        MagicMock(vid=0x1234, pid=0x5678, device="COM4"),
+    ]
```

Comment:
can we put this in a fixture? or is there a reason that we're mocking in a test function?

Reply:
but also since i know that you have experiments to run in the near future, i'd be ok pushing this off if we just raise an issue to remind ourselves to do it

Reply:
Oh yea I was playing around with COM/pids while writing this and just left it that way. fixtures will scale better.

Reply:
I was initially thinking about full mock, but this device literally has no response and accepts commands in one-way. The response comes through parameter changes coming in from the `streamdaq` module, and I don't have time to integrate these so ended up this way. But makes sense to keep it as an issue in mind

Reply:
not mind probabl ygithub

Reply:
> I was initially thinking about full mock, but this device literally has no response and accepts commands in one-way

hahaha ok, that makes sense. silent device makes for uninteresting mocks. yeah let's raise it as an issue to track needing to do once the system is more mature/we're closer to release and we want to do formal verification. a fixture that bundles together monkeypatching the relevant methods/attrs like you have already done here would be nice, as you say, for scale, but understood why simulating the device behavior here might be less than enlightening :). 

something like this that just stashes the values passed might be nice given the pattern of some of these tests? https://github.com/Aharoni-Lab/miniscope-io/blob/56b32b7345698c004c1170aac664276ff7836de7/miniscope_io/devices/mocks.py#L74

but also w/e is fine.

Reply:
Makes sense, but I think I'll skip this for now as there's much higher priority stuff in my task queue. I'll just change to parameterized fixtures for this PR.

Reply:
github needs a better interface for "just talking, not trying to hold up work, merge whenever" bc that's where i've been at for awhile on this one

Reply:
Yea I was going to merge whenever I do the fixture stuff, so not really held by your comments, but there could be a chat interface.